lib/kubeclient/compatibility_3_1.rb

```diff
@@ -0,0 +1,68 @@
+module Kubeclient
+  module Common
+    # Backward compatibility for 3.1
+    class Compatibility31
```

Comment:
@moolitayer @simon3z I would like to go with @cben's suggestion `MissingKindCompatibility` since we are not sure yet that kind is missing in k8s 1.1, are you OK with that?